File: src/services/TradeResult.cs
namespace StarSmugglerGo.Services;

public sealed class TradeResult
{
    public bool Succeeded { get; init; }
    public string Message { get; init; } = string.Empty;

    public static TradeResult Success(string message)
    {
        return new TradeResult
        {
            Succeeded = true,
            Message = message,
        };
    }

    public static TradeResult Failure(string message)
    {
        return new TradeResult
        {
            Succeeded = false,
            Message = message,
        };
    }
}

File: internal/services/trade_result.go
package services

type TradeResult struct {
	Succeeded bool
	Message   string
}

func SuccessfulTrade(message string) TradeResult {
	return TradeResult{
		Succeeded: true,
		Message:   message,
	}
}

func FailedTrade(message string) TradeResult {
	return TradeResult{
		Succeeded: false,
		Message:   message,
	}
}
